src/timezones/rounding.jl

```diff
@@ -0,0 +1,146 @@
+function Base.floor(zdt::ZonedDateTime, p::Dates.DatePeriod)
+    return ZonedDateTime(floor(localtime(zdt), p), zdt.timezone)
+end
+
+function Base.floor(zdt::ZonedDateTime, p::Dates.TimePeriod)
+    # Rounding is done using the current fixed offset to avoid transitional ambiguities.
+    dt = floor(localtime(zdt), p)
+    utc_dt = dt - zdt.zone.offset
+    return ZonedDateTime(utc_dt, zdt.timezone; from_utc=true)
+end
+
+function Base.ceil(zdt::ZonedDateTime, p::Dates.DatePeriod)
+    return ZonedDateTime(ceil(localtime(zdt), p), zdt.timezone)
+end
+
+#function Base.Dates.floorceil(zdt::ZonedDateTime, p::Dates.DatePeriod)
+    #return floor(zdt, p), ceil(zdt, p)
+#end
+
+"""
+    floor(zdt::ZonedDateTime, p::Period) -> ZonedDateTime
```

Comment:
Multiple signatures are supported. When working with Base you just need to make sure that the signatures you supply in the docstring match that of the manual.